### PR TITLE
added some classes which are connected with user entity and lo…

### PR DESCRIPTION
**Pull request by Dimitriy1**
…gin an registration pages

### CODE REVIEW

File: src/main/java/mate/academy/spring/controller/DeveloperController.java [hunk 1]
```diff
@@ -30,23 +31,56 @@ public String index() {
...
-        modelMap.addAttribute("developer",developerOutput);
+        modelMap.addAttribute("developer", developerOutput);
 
         return "/developers/info";
     }
+
+    @RequestMapping(value = "/developer/showAllDevelopers", method = RequestMethod.GET)
+    public String getAllDevelopers(ModelMap modelMap) {
+        Set<Developer> developers = developerService.findAllDevelopers();
+        Set<DeveloperOutput> developerOutputs = new LinkedHashSet<>();
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/spring/controller/DeveloperController.java]: use stream api

File: src/main/java/mate/academy/spring/model/User.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Column;
+
+@Entity
+@Table(name = "user")
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/spring/model/User.java]: you wrote Data
should not write getter and setter

File: src/main/java/mate/academy/spring/model/dto/DeveloperInput.java [hunk 3]
```diff
@@ -1,49 +1,20 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Data;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/spring/model/dto/DeveloperInput.java]: you wrote Data
should not write getter and setter

File: src/main/java/mate/academy/spring/model/dto/DeveloperOutput.java [hunk 4]
```diff
@@ -1,49 +1,20 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Data;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/spring/model/dto/DeveloperOutput.java]: you wrote Data
should not write getter and setter

File: src/main/java/mate/academy/spring/model/dto/UserLoginInput.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Data;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/spring/model/dto/UserLoginInput.java]: you wrote Data
should not write getter and setter

File: src/main/java/mate/academy/spring/model/dto/UserRegistrationInput.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.Data;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/spring/model/dto/UserRegistrationInput.java]: you wrote Data
should not write getter and setter

File: src/main/webapp/WEB-INF/pages/developers/putDeveloper.jsp [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+                    " link: <a href='" + url + "'>" + url + "</a>");
+            },
+            error: function (xhr, status, error) {
+                alert(error);
+            }
+        });
+    });
+</script>
+</body>
+</html>
```
Comment (BondarenkoJek) [inline src/main/webapp/WEB-INF/pages/developers/putDeveloper.jsp]: after last bracers should be one empty line

File: src/main/java/mate/academy/spring/controller/UserController.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @RequestMapping(value = "/register", method = RequestMethod.POST)
+    public void register(ModelMap modelMap, UserRegistrationInput userRegistrationInput) {
+    }
+
+    @RequestMapping(value = "/login", method = RequestMethod.POST)
+    public void login(ModelMap modelMap, UserLoginInput userLoginInput) {
+    }
+
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/spring/controller/UserController.java]: delete an empty line

File: src/main/java/mate/academy/spring/service/UserServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
+package mate.academy.spring.service;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component("userService")
```
Comment (BondarenkoJek) [inline src/main/java/mate/academy/spring/service/UserServiceImpl.java]: should use annotation Service

File: src/main/java/mate/academy/spring/controller/DeveloperController.java [hunk 10]
```diff
@@ -9,12 +9,14 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.ModelMap;
-import org.springframework.web.bind.annotation.ModelAttribute;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.*;
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/controller/DeveloperController.java]: we don't using * import

File: src/main/java/mate/academy/spring/controller/DeveloperController.java [hunk 11]
```diff
@@ -30,23 +32,55 @@ public String index() {
...
+        modelMap.addAttribute("developer", developerOutput);
 
         return "/developers/info";
     }
 
     @RequestMapping(value = "/developer/information", method = RequestMethod.GET)
     public ModelAndView student() {
-        return new ModelAndView("/developers/form", "addDeveloper", new DeveloperInput());
+        return new ModelAndView("/developers/form",
+                "addDeveloper", new DeveloperInput());
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/controller/DeveloperController.java]: @Dimitriy1 do you required new DeveloperInput() here? Maybe you should rename endpoint to /developer/create ?

File: src/main/java/mate/academy/spring/controller/UserController.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+@Controller
+@Log4j2
+@RequestMapping("/users")
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/controller/UserController.java]: Why users, not Auth?

File: src/main/java/mate/academy/spring/controller/UserController.java [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    @RequestMapping(value = "/login", method = RequestMethod.GET)
+    public String login() {
+        log.info("a test message");
+        return "/users/logInForm";
+    }
+
+    @RequestMapping(value = "/register", method = RequestMethod.POST)
+    public void register(ModelMap modelMap, UserRegistrationInput userRegistrationInput) {
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/controller/UserController.java]: Use log4j and log userRegistrationInput.toString() inside the method

File: src/main/java/mate/academy/spring/controller/DeveloperController.java [hunk 14]
```diff
@@ -39,7 +44,7 @@ public String getDeveloperInfo(@PathVariable Long developerId, ModelMap modelMap
 
     @RequestMapping(value = "/developer/information", method = RequestMethod.GET)
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/controller/DeveloperController.java]: @Dimitriy1 is information endpoint ok for user creating???

**Pull request merged**